src/meltano/cli/config.py

```diff
@@ -378,8 +381,38 @@ def set_(
         setting_name = tuple(setting_name[0].split("."))
 
     interaction = InteractiveConfig(ctx=ctx, store=store, extras=False)
+
     if interactive:
-        interaction.configure_all()
+        if sys.stdin.isatty():
+            # Interactive stdin flow
+            interaction.configure_all()
+        else:
```

Comment:
We got rid of the breaking change that replaced `--interactive` with `--stdin` but it feels a bit weird having to add `--interactive` to accept stdin.

Wdyt of having a `--from-file` option that accepts a filename we read the contents of:

```python
    if interactive:
        interaction.configure_all()
        ctx.exit()

    if from_file:
        setting_name += (value,)
        value = from_file.read()

    interaction.set_value(setting_name=setting_name, value=value, store=store)
```

This feels more complete, and we get `-` support out of the box:

> Command line tools are more fun if they work with files the Unix way, which is to accept `-` as a special file that refers to stdin/stdout.

https://click.palletsprojects.com/en/8.1.x/arguments/#file-arguments

Reply:
I checked the link, In this `-` would tell meltano that standard input will be provide but we'll still get `-` as an argument and we'll have to again remove it from setting_name. Moreover if we are not passing any file and just wan to use this `poetry run meltano --cwd ../shubh config meltano set cli log_level info`, it will still expect a file and in this case will consider info as a file.

Now either we should pass the the file as a value to this flag or the best is to just allow user to stream input using < file_location which we are doing right now, but we'll just rename the flag.

Also this --from-file can be confusing as it can convey that user has to provide file name of streaming a file, should we use `--stdin-from-file`. Now user can either pipe the input or just stream it from a file if they are using this flag.

wdyt?

Reply:
> I checked the link, In this `-` would tell meltano that standard input will be provide but we'll still get `-` as an argument and we'll have to again remove it from setting_name.

click supports `-` as a special value for `File` parameters, so there's no need for that.

> Also this --from-file can be confusing as it can convey that user has to provide file name of streaming a file, should we use `--stdin-from-file`. Now user can either pipe the input or just stream it from a file if they are using this flag.

I'm sorry I don't see how it'd cause that confusion.

```bash
# Configure a plugin with the contents of a file
meltano config tap-example set private_key --from-file private_key.txt

# Configure a plugin from stdin
uuidgen | meltano config tap-example set salt --from-file -
```

Reply:
These are 2 different things, This `@click.argument('input', type=click.File('rb'))` will only support the later one cause in 1st we are just providing file as an argument and it requires byte stream.

I'll say lets not make it complex and support this instead. Please share your thoughts


```
meltano config tap-example set -from-file cli < file.txt

# Configure a plugin from stdin
cat file.txt | meltano config tap-example set --from-file cli
```

and we'll just read the stdin in both cases

Reply:
@edgarrmondragon thoughts?

Reply:
@edgarrmondragon thoughts?

Reply:
Ok, so here's what I mean:

1. We change the scope of this issue to something like `feat: Support setting config from a file input and stdin`
2. We use `@click.argument('--from-file', type=click.File('r'))` to accept a file (in text mode). This lets us accept `-` as a filename to signify capturing stdin.
3. If `--from-file` is not null, we process the setting name to append the tail value

Reply:
Done